go/private/context.bzl

```diff
@@ -70,6 +70,8 @@ _GoContextData = provider()
 
 _COMPILER_OPTIONS_BLACKLIST = {
     "-fcolor-diagnostics": None,
+    "-fdiagnostics-color": None,
+    "-fmax-errors=20": None,
```

Comment:
Why exclude this flag? And why 20 specifically?

Reply:
In our code base, we add -fmax-errors=20.  Generally only the first couple of errors are actually helpful when something fails to compile.  cgo tests the compiler with a file which has more than 20 errors, so the presence of this flag drops errors it is expecting and fails the build.

Currently your filter function doesn't support a startswith list.  I can add one, though it'll be slower to parse.  I could also find the limit and add -fmax-errors={0..N} to filter those out.  What would you prefer?

Reply:
I'd suggest adding a `_match_option` function:

```
def _match_option(opt, pattern):
    if pattern.endswith("="):
        return opt.startswith(pattern)
    else:
        return opt == pattern
```

Then using that as a predicate in `_filter_options`:

```
def _filter_options(options, blacklist):
    return [
        option
        for option in options
        if not any([_match_option(option, pattern) for pattern in blacklist])
    ]
```

A quick comment that `-fmax-errors=N` breaks cgo would be helpful.

Reply:
Sweet, done.